src/nebula/NebulaAction.cpp

```diff
@@ -344,6 +358,173 @@ ResultCode CheckLeadersAction::checkResp(const ExecutionResponse& resp) const {
+        auto ret = checkResp(resp);
+        if (ret == ResultCode::ERR_FAILED_NO_RETRY) {
+            return ret;
+        }
+        if (ret == ResultCode::OK) {
+            if (resultVarName_.empty()) {
+                return ResultCode::OK;
+            } else {
+                ret = checkLeaderDis(resp);
+                if (ret == ResultCode::ERR_FAILED_NO_RETRY) {
+                    return ret;
+                }
+                if (ret == ResultCode::OK) {
+                    ctx_->exprCtx.setVar(resultVarName_, std::move(restult_));
+                    return ret;
+                }
+            }
+        }
+        LOG(ERROR) << "Execute " << cmd << " failed, retry " << retry
+                   << " after " << retry << " seconds...";
+        sleep(retry);
+    }
+    return ResultCode::ERR_FAILED;
+}
+
+ResultCode SetFlagAction::buildCmd() {
+    folly::toLowerAscii(layer_);
+    folly::toLowerAscii(name_);
+    folly::toLowerAscii(value_);
+    if (layer_ != "storage" && layer_ != "meta" && layer_ != "graph") {
+       return ResultCode::ERR_FAILED;
+    }
+
+    cmd_ ="update configs ";
+    cmd_ += layer_;
+    cmd_ += ":";
+    if (name_ == "disable_auto_compactions")  {
+        cmd_ += "rocksdb_column_family_options = { disable_auto_compactions = ";
+        if (value_ != "true" && value_ != "false") {
+            LOG(ERROR) << "Failed to get rocksdb_column_family_options value";
+            return ResultCode::ERR_FAILED;
+        }
+        cmd_ += value_;
+        cmd_ += " } ";
+    } else if (name_ == "wal_ttl") {
+        try {
+            auto ttlVal = folly::to<int32_t>(value_);
+            value_ = folly::to<std::string>(ttlVal);
+        } catch (const folly::ConversionError& e) {
+            LOG(ERROR) << "Failed to get wal_ttl value";
+            return ResultCode::ERR_FAILED;
+        }
+        cmd_ += name_;
+        cmd_ += " = ";
+        cmd_ += value_;
+    }
+    return ResultCode::OK;
+}
+
+ResultCode SetFlagAction::doRun() {
+    CHECK_NOTNULL(client_);
+    auto ret = buildCmd();
+    if (ret != ResultCode::OK) {
+        return ret;
+    }
+
+    auto cmd = command();
+    LOG(INFO) << "Send " << cmd << " to graphd";
+    ExecutionResponse resp;
+    int32_t retry = 0;
+    while (++retry < retryTimes_) {
+        client_->execute(cmd, resp);
+        LOG(INFO) << "Execute " << cmd << " successfully!";
+        ret = checkResp(resp);
+        if (ret == ResultCode::OK || ret == ResultCode::ERR_FAILED_NO_RETRY) {
+            return ret;
+        }
+        LOG(ERROR) << "Execute " << cmd << " failed, retry " << retry
+                   << " after " << retry << " seconds...";
+        sleep(retry);
+    }
+    return ResultCode::ERR_FAILED;
+}
+
+ResultCode CompactionAction::checkResp(const ExecutionResponse& resp) const {
+    auto errorCode = resp.get_error_code();
+    if (errorCode == ::nebula::graph::cpp2::ErrorCode::SUCCEEDED) {
```

Comment:
submit job compaction will be returned immediately,  So it will be always Succeeded?

Reply:
If add job is successful (in submit job compaction), return succeeded, otherwise it may fail